pylib/cqlshlib/tracing.py

```diff
@@ -33,7 +33,15 @@ def print_trace_session(shell, session, session_id, partial_session=False):
     except TraceUnavailable:
         shell.printerr("Session %s wasn't found." % session_id)
     else:
-        print_trace(shell, trace)
+        print_trace(shell, tracing_mode, trace)
+
+
+def print_trace(shell, tracing_mode, trace):
+    assert tracing_mode in {'full', 'compact'}
+    if tracing_mode == 'full':
+        print_trace_full(shell, trace)
+    else:
+        print_trace_compact(shell, trace)
```

Comment:
Nit: `print_trace_full(shell,trace) if tracing_mode == 'full' else print_trace_compat(shell, trace)` is probably slightly preferred.

Even better might be something like:

```python

trace_fn = print_trace_full if tracing_mode == 'full' else print_trace_compat
trace_fn(shell, trace)
```

Reply:
wfm, altho this is more fragile if they evolve different parameters later